service/controller/resource/statuscondition/create.go

```diff
@@ -186,103 +250,58 @@ func (r *Resource) computeCreateClusterStatusConditions(ctx context.Context, cl
 			readyWorkerReplicas += int(md.Status.ReadyReplicas)
 		}
 	}
+	return readyWorkerReplicas == desiredWorkerReplicas
+}
 
-	// After initialization the most likely implication is the tenant cluster
-	// being in a creation status. In case no other conditions are given and no
-	// versions are set, we set the tenant cluster status to a creating
-	// condition.
-	{
-		notCreating := !status.HasCreatingCondition()
-		conditionsEmpty := len(status.Conditions) == 0
-		versionsEmpty := len(status.Versions) == 0
-
-		if notCreating && conditionsEmpty && versionsEmpty {
-			status.Conditions = status.WithCreatingCondition()
-			r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("setting %#q status condition", infrastructurev1alpha2.ClusterStatusConditionCreating))
-			r.event.Emit(ctx, &cl, "ClusterInCreation", fmt.Sprintf("cluster creation is in condition %s", infrastructurev1alpha2.ClusterStatusConditionCreating))
-		}
+func allNodesHaveVersion(nodes []corev1.Node, version string, providerOperatorVersionLabel string) bool {
```

Comment:
This function checks if all nodes have versions (it has been there before and is unchanged)